src/player_menu.js

```diff
@@ -33,77 +41,62 @@ export async function display_menu({ client, world, position, background }) {
   )
 
   const { canvas } = create_screen_canvas(world.screens.player_screen)
-
   const ctx = canvas.getContext('2d')
-  // load image
-  const myimg = await loadImage(background)
-  ctx.drawImage(myimg, 0, 0, 128 * 6, 128 * 4)
 
+  const myimg = await loadImage(type?.image)
+  ctx.drawImage(myimg, 0, 0, 128 * 6, 128 * 4)
+  
   spawn_screen(
     { client, world },
     {
-      screen_id: 'player_screen',
+      screen_id: type.screen_id,
       position: screen_positon,
       direction: right.clone(),
     }
   )
-
   update_screen(
     { client, world },
-    { screen_id: 'player_screen', new_canvas: canvas, old_canvas: null }
-  )
-
-  return { screen_positon, direction, right }
-}
-
-export function menu_clone(
-  client,
-  cloneId,
-  inventory,
-  { screen_positon, right, direction }
-) {
-  const wieldedArmor = inventory.slice(5, 9).reverse() // reversed to fit armor stand slots
-  const equipement = [null, null, ...wieldedArmor] // TODO: change null to equiped weapon
-
-  create_player(
-    client,
-    cloneId,
-    equipement,
-    screen_positon
-      .clone()
-      .add(right.scaled(1.5))
-      .offset(0, -1.75, 0)
-      .add(direction.scaled(1.4)),
-    (-direction_to_yaw_pitch(direction).yaw + (50 / 360) * 255) % 127,
-    -direction_to_yaw_pitch(direction).yaw % 127
+    { screen_id: type.screen_id, new_canvas: canvas, old_canvas: null }
   )
-}
-
-export function clone_background(
-  client,
-  world,
-  { screen_positon, right, direction }
-) {
-  const { canvas: canvas2 } = create_screen_canvas(world.screens.player_screen2)
-  const ctx2 = canvas2.getContext('2d')
-  ctx2.fillStyle = '#E3CFAF'
-  ctx2.fillRect(0, 70, 128 * 3, 128 * 4)
+  if (inventory !== null && clone_id !== null) {
```

Comment:
if clone id can never be 0 then:
```suggestion
  if (inventory && clone_id) {
```